authentication/src/test/java/io/camunda/authentication/filters/SessionAuthenticationRefreshFilterTest.java

```diff
@@ -31,172 +32,241 @@
+      final Instant newRefreshTime = (Instant) session.getAttribute(LAST_REFRESH_ATTR);
+      assertThat(newRefreshTime).isAfter(Instant.now().minusMillis(refreshInterval.toMillis()));
+    }
+
+    private void setupAuthentication() {
+      final var mockAuthentication = createMockAuthentication();
+      final var testAuthentication = new TestingAuthenticationToken(mockAuthentication, null);
+      testAuthentication.setAuthenticated(true);
+      SecurityContextHolder.getContext().setAuthentication(testAuthentication);
+      when(authenticationProvider.getCamundaAuthentication()).thenReturn(mockAuthentication);
+    }
+
+    private CamundaAuthentication createMockAuthentication() {
+      return new CamundaAuthentication(
+          TestUserDetailsService.DEMO_USERNAME,
+          "",
+          Collections.emptyList(),
+          Collections.emptyList(),
+          Collections.emptyList(),
+          Collections.emptyList(),
+          new HashMap<>());
+    }
+
+    private void setSessionRefreshAttribute(
+        final MockHttpSession session, final Duration refreshInterval) {
+      final Instant lastRefreshRef = Instant.now().minus(refreshInterval);
+      session.setAttribute(LAST_REFRESH_ATTR, lastRefreshRef);
+      session.setAttribute(LAST_REFRESH_ATTR + "_LOCK", session.getId() + "LOCK");
     }
 
-    assertThat(successfulRequests.get()).isEqualTo(threads.length);
-    verify(authenticationProvider, times(1)).refresh();
-    final Instant newRefreshTime = (Instant) session.getAttribute(LAST_REFRESH_ATTR);
-    assertThat(newRefreshTime).isAfter(Instant.now().minusMillis(refreshInterval.toMillis()));
+    private SendMultipleRequest getSendMultipleRequest(
+        final MockHttpSession session,
+        final CamundaAuthenticationProvider authenticationProvider,
+        final MockMvcTester mockMvcTester)
+        throws InterruptedException {
+      doAnswer(
+              invocation -> {
+                Thread.sleep(100);
+                return null; // because it's a void method
+              })
+          .when(authenticationProvider)
+          .refresh();
+      final AtomicInteger successfulRequests = new AtomicInteger(0);
+      final Runnable requestTask =
+          () -> {
+            try {
+              setupAuthentication();
+              final MvcTestResult testResult =
+                  mockMvcTester
+                      .get()
+                      .session(session)
+                      .uri("https://localhost" + TestApiController.DUMMY_WEBAPP_ENDPOINT)
+                      .exchange();
+              assertThat(testResult).hasStatusOk();
+              successfulRequests.incrementAndGet();
+            } catch (final Exception e) {
+              throw new RuntimeException(e);
+            }
+          };
+
+      final Thread[] threads = new Thread[5];
+      for (int i = 0; i < threads.length; i++) {
+        threads[i] = new Thread(requestTask);
+        threads[i].start();
```

Comment:
I tried running the tests locally and they are still flaky. I put the testing on repeat and increased thread count to 100 to induce failure faster. There seems to be an occasional issue (ca 5% of runs) with Mockito stubbing where it throws an exception due to `refresh()` not being stubbed properly. There could be a hidden underlying issue but I could not find it quickly.
I tried using a `CountDownLatch` and `ExecutorService` but that did not help.
The only thing that made an effect is separating the spin of the Threads a bit, even by sleeping for 1 ms.
Therefore I suggest we add the following line to stabilize the test, unless someone has the time to dig more into root causes now :) 
```suggestion
        threads[i].start();
        Thread.sleep(10);
```